packages/happy-dom/src/xml-parser/XMLParser.ts

```diff
@@ -51,7 +52,14 @@ export default class XMLParser {
 
 				if (parent && match.index !== lastTextIndex) {
 					const text = data.substring(lastTextIndex, match.index);
-					this.appendTextAndCommentNodes(document, parent, text);
+					if (
+						parent.nodeType === Node.ELEMENT_NODE &&
+						PlainTextElements.includes((<IElement>parent).tagName.toLowerCase())
```

Comment:
I think we could store the parent tag name to improve performance a bit as we only need to convert to lower case once.

We also need to check that the parent in fact has a tag name as it can also be a Document or DocumentFragment.

Example:
```javascript
const parentTagName = <IElement>parent).tagName ? <IElement>parent).tagName.toLowerCase() : null;
```

We can also then add a check for if the `parentTagName` is set to the if statement at row 57.

Reply:
Sure, excellent suggestions, and thank you for your quick response 👍